### ISSUE TITLE
Fish-eye shader: distortion in only x or y direction?

### ISSUE THREAD

**Issue by ArcadeHall** [exeldro/obs-lua#3]
I have a question about the fish-eye shader: is it possible to add a barrel distortion which only works in x or y direction of the image? I have looked at the formulas in the fish-eye shader, but that's over my head. I have a webcam which has a fish-eye distortion which can be corrected by a negative power in the fish-eye shader. But still the image looks a bit like this: )==(. I only need some correction in the x direction, but not y, since the lines are straight vertically. Can you give me a formula for this perhaps?

**Comment by exeldro** [comment]
can you test if 4582bad5ae68b281abc3b6277c4e18e553b629f3 works for your use case?

**Comment by ArcadeHall** [comment]
Yes, now I can correct the X and Y distortion seperately, thank you very much! 
Unfortunately now another distortion is visible: not a barrel distortion, but a cylindrical distortion. It looks a like reading a letter through a long drink glass laid down (horizontal, I hope it makes sense). Do you have a formula for cylindrical distortion (x and y separate if possible)?
